File: python/sublist/sublist.py
SUBLIST = 0
SUPERLIST = 1
EQUAL = 2
UNEQUAL = 3


def check_lists(l1, l2):
    if l1 == l2:
        return EQUAL
    elif l1 in each_cons(l2, len(l1)):
        return SUBLIST
    elif l2 in each_cons(l1, len(l2)):
        return SUPERLIST
    else:
        return UNEQUAL


# Not the most efficent
def each_cons(lst, size):
    return [lst[i: i + size] for i in range(len(lst) - size + 1)]

File: go/sublist/sublist.go
package sublist

import (
	"reflect"
	"strconv"
	"strings"
)

type Relation string

func Sublist(a []int, b []int) Relation {
	if reflect.DeepEqual(a, b) {
		return "equal"
	} else if isSublist(a, b) {
		return "sublist"
	} else if isSublist(b, a) {
		return "superlist"
	} else {
		return "unequal"
	}
}

// isSublist returns true if a is a sublist of b
func isSublist(a []int, b []int) bool {
	stringA := getString(a)
	stringB := getString(b)
	return strings.Contains(stringB, stringA)
}

func getString(a []int) string {
	result := []string{}
	for _, val := range a {
		str := strconv.Itoa(val)
		result = append(result, str)
	}
	return strings.Join(result, ",")
}
